clang/include/helper.hpp

```diff
@@ -66,4 +68,11 @@ struct CopyPtr {
 	const T &operator*() const { return *this->ptr; }
 };
 
+// compatibility for C++11
+template< typename T, typename... Args >
+std::unique_ptr<T> make_unique(Args&&... args)
```

Comment:
In the future I don't think we'll want to support C++11 compilers.  When I started this I think I had C++14-ish in mind, I think it's safe nowadays to move on to C++17.  This will also enable us to use `std::variant` and `std::optional` in certain places.